cmd/syncer/root.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2021 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"path"
+	"syscall"
+
+	nscontroller "github.com/kcp-dev/kcp/pkg/reconciler/namespace"
+	"github.com/kcp-dev/kcp/pkg/syncer"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+)
+
+const numThreads = 2
+
+var (
+	fromKubeconfig      string
+	fromClusterName     string
+	toKubeconfig        string
+	toContext           string
+	pclusterID          string
+	syncedResourceTypes []string
+)
```

Comment:
Please move this into an `Options` struct - see https://github.com/kcp-dev/kcp/blob/ba104a795fa57e7f452a91b5fb30f96463aa1599/cmd/ingress-controller/main.go#L122-L150 for a good reference example.

Reply:
@ncdc, should I keep the Logs field as well in the Syncer Options struct? Will there be any future prospects to it? WDYT

Reply:
Yes, please. It configures `--v` and related flags.